tests/runtime/main.rs

```diff
@@ -453,5 +489,225 @@ fn tests(name: &str, dir_name: &str) -> Result<Vec<PathBuf>> {
         result.push(component_path);
     }
 
+    #[cfg(feature = "csharp")]
+    for path in c_sharp.iter() {
+        println!("running for {}", path.display());
+        let world_name = &resolve.worlds[world].name;
+        let out_dir = out_dir.join(format!("csharp-{}", world_name));
+        drop(fs::remove_dir_all(&out_dir));
+        fs::create_dir_all(&out_dir).unwrap();
+
+        for csharp_impl in &c_sharp {
+            fs::copy(
+                &csharp_impl,
+                &out_dir.join(csharp_impl.file_name().unwrap()),
+            )
+            .unwrap();
+        }
+
+        let snake = world_name.replace("-", "_");
+
+        let assembly_name = format!(
+            "csharp-{}",
+            path.file_stem().and_then(|s| s.to_str()).unwrap()
+        );
+
+        let out_wasm = out_dir.join(&assembly_name);
+
+        let mut files = Default::default();
+        let mut opts = wit_bindgen_csharp::Opts::default();
+        if let Some(path) = path.file_name().and_then(|s| s.to_str()) {
+            if path.contains("utf16") {
+                opts.string_encoding = wit_component::StringEncoding::UTF16;
+            }
+        }
+        opts.build().generate(&resolve, world, &mut files).unwrap();
+
+        fs::write(
+            out_dir.join("nuget.config"),
+            r#"<?xml version="1.0" encoding="utf-8"?>
+        <configuration>
+            <config>
+                <add key="globalPackagesFolder" value=".packages" />
+            </config>
+            <packageSources>
+            <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+            <clear />
+            <add key="nuget" value="https://api.nuget.org/v3/index.json" />
+            <!--<add key="dotnet-experimental" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-experimental/nuget/v3/index.json" />-->
+            <add key="dotnet-experimental" value="C:\github\runtimelab\artifacts\packages\Debug\Shipping" />
```

Comment:
this looks like it is the project for the llvm approach?  Since we want to cover both will we need to create a sperate solution file for the mono?

Reply:
Good point, yes, we will need a different project file for mono, so I guess that would mean, as far as running the tests, that we should have a `csharp-naot` feature and a `csharp-mono` feature.